src/lib/mina_base/test/account/account.ml

```diff
@@ -0,0 +1,270 @@
+          let open Global_slot in
+          sub max_value timing_rec.vesting_period
+          |> Option.value_map ~default:vesting_end ~f:(min vesting_end)
+        in
+        let%map global_slot =
+          Global_slot.gen_incl timing_rec.cliff_time max_slot
+        in
+        (timing, timing_rec, global_slot))
+        ~f:(fun (timing, timing_rec, global_slot) ->
+          let vesting_period_later =
+            Global_slot.add global_slot timing_rec.vesting_period
+          in
+          let min_bal_at_slot = min_balance_at_slot timing_rec ~global_slot in
+          let min_bal_later =
+            min_balance_at_slot timing_rec ~global_slot:vesting_period_later
+          in
+          [%test_eq: Balance.t] min_bal_later
+            Balance.(
+              Option.value ~default:zero
+              @@ (min_bal_at_slot - timing_rec.vesting_increment)) )
+
+    let%test_unit "Incremental balance between slots before cliff is 0." =
+      Quickcheck.test
+        (let open Quickcheck.Generator.Let_syntax in
+        let%bind timing = gen_timing in
+        let max_slot =
+          Global_slot.(sub timing.cliff_time (of_int 1))
+          |> Option.value ~default:Global_slot.zero
+        in
+        let%bind slot1 = Global_slot.(gen_incl zero max_slot) in
+        let%map slot2 = Global_slot.gen_incl slot1 max_slot in
+        (timing, slot1, slot2))
+        ~f:(fun (timing, start_slot, end_slot) ->
+          [%test_eq: Balance.t] Balance.zero
+            (incr_bal_between timing ~start_slot ~end_slot) )
+
+    let%test_unit "Incremental balance between slots after vesting finished is \
+                   0." =
+      Quickcheck.test
+        (let open Quickcheck.Generator.Let_syntax in
+        let%bind timing = gen_timing in
+        let vesting_end =
+          Account.timing_final_vesting_slot (Account.Timing.of_record timing)
+        in
+        let%bind slot1 = Global_slot.(gen_incl vesting_end max_value) in
+        let%map slot2 = Global_slot.(gen_incl slot1 max_value) in
+        (timing, slot1, slot2))
+        ~f:(fun (timing, start_slot, end_slot) ->
+          [%test_eq: Balance.t] Balance.zero
+            (incr_bal_between timing ~start_slot ~end_slot) )
+
+    let%test_unit "Incremental balance where end is before start is 0." =
+      Quickcheck.test
+        (let open Quickcheck.Generator.Let_syntax in
+        let%bind timing = gen_timing in
+        let%bind slot1 = Global_slot.gen in
+        let%map slot2 = Global_slot.(gen_incl zero slot1) in
+        (timing, slot1, slot2))
+        ~f:(fun (timing, start_slot, end_slot) ->
+          [%test_eq: Balance.t] Balance.zero
+            (incr_bal_between timing ~start_slot ~end_slot) )
+
+    let%test_unit "Incremental balance during vesting is a multiple of \
+                   vesting_increment." =
+      Quickcheck.test
+        (let open Quickcheck in
+        let open Generator.Let_syntax in
+        let%bind timing =
+          Generator.filter
```

Comment:
This filter should be removed now that the quickcheck generator has been updated.

Reply:
So I'll remove the Generator.filter and ~f argument. It will just be 
`let%bind timing = Account.gen_timing_at_least_one_vesting_period Balance.max_int`

Reply:
When I run the tests for this it fails. Leaving it as it was, the test passes.

Reply:
Hmm, that means the generator implementation isn't correct yet then.

Reply:
The error if it helps was ```
("Bigint.gen_uniform_incl: bounds are crossed" (lower_bound 1837293401)
  (upper_bound 1837293400)).
```